### ISSUE TITLE
[win32] if_addrs::get_if_addrs doesn't list link-local ipv6 addresses

### ISSUE THREAD

**Issue by nvandamme** [librespot-org/libmdns#52]
Using the embeded call of  ```rust if_addrs::get_if_addrs()``` doesn't list link-local ipv6 addresses on `win32`.
The implementation proposed by the crate `local-ip-address` resolve this.

Proposing to replace ```rust if_addrs::get_if_addrs()``` call with ```rust local-ip-address::list_afinet_netifas()```

Example: 

```rust
let if_addrs: Vec<std::net::IpAddr> = local-ip-address::list_afinet_netifas()
    .expect("Cannot get local IP addresses")
    .into_iter()
    .map(|(_, ip)| ip)
    .collect::<Vec<std::net::IpAddr>>();
```

**Comment by nvandamme** [comment]
Might be a good idea to unsure smotth IPv6 publishing by binding the right interface using it's index ? 

There is a new crate that is lisitng iface indexes (unix & win32): https://github.com/librespot-org/libmdns/issues/53#issuecomment-2038414189

**Comment by willstott101** [comment]
(finally) released the suggested fix in 0.9.0
